src/api/types.rs

```diff
@@ -1,14 +1,71 @@
+extern crate chrono;
 
-
-#[derive(Queryable, Serialize, Deserialize)]
+#[derive(Queryable, Insertable, Serialize, Deserialize)]
+#[table_name="test_names"]
 pub struct TestName {
-    pub id: i32,
     pub name: String,
 }
 
+#[derive(Queryable, Insertable)]
+#[table_name="artifact_names"]
+pub struct ArtifactName {
+	pub name: String,
+}
+
+#[derive(Queryable, Insertable)]
+#[table_name="version"]
+pub struct Version {
+	pub id: i32,
+	pub major: String,
+	pub minor: Option<String>,
+	pub patch: Option<String>,
+	pub build: Option<String>,
+}
+
+#[derive(Queryable, Insertable)]
+#[table_name="test_info"]
+pub struct TestInfo {
+	pub id: i32,
+	pub test_name: String,
```

Comment:
so you haven't figured out how to link the test_name, version, etc to the table?